src/frontend/src/handler/create_source.rs

```diff
@@ -667,7 +661,7 @@ pub fn handle_addition_columns(
             );
         }
         let col = build_additional_column_desc(
-            latest_col_id.next(),
+            ColumnId::placeholder(),
```

Comment:
The `latest_col_id.next()` logic was wrong because it always returns the same value in this `while` loop given `latest_col_id` is not `mut`. The actual column id is assigned in the caller after calling `handle_addition_columns`.